lib/utils.coffee

```diff
@@ -305,7 +294,7 @@ exports.requestAsync = (options, retriesRemaining = undefined) ->
 	requestTime = new Date()
 	p = fetch(url, opts)
 	if timeout
-		p = timeoutPromise(timeout, p)
+		p = p.timeout(timeout)
```

Comment:
Just for reference, we could just add `.catch Promise.TimeoutError, -> throw new Error('timeout')`